pkg/restapi/device/operations.go

```diff
@@ -231,6 +307,50 @@ func (o *Operation) getUserData(w http.ResponseWriter, r *http.Request) (userDat
 	return userData, true
 }
 
-func marshallResponse(value interface{}) ([]byte, error) {
-	return json.Marshal(value)
+func (o *Operation) saveDeviceInfo(device *Device) error {
+	deviceBytes, err := json.Marshal(device)
+	if err != nil {
+		return fmt.Errorf("failed to marshall the device data %s", err)
+	}
+
+	err = o.store.transient.Put(device.ID, deviceBytes)
```

Comment:
device info shouldn't be saved just to transient storage

Reply:
done moved to permanent storage